nautobot/extras/models/jobs.py

```diff
@@ -807,6 +819,61 @@ def log(
             logger.log(log_level, message)
 
 
+#
+# Job Button
+#
+
+
+@extras_features("graphql")
+class JobButton(BaseModel, ChangeLoggedModel, NotesMixin):
+    """
+    A predefined button that includes all necessary information to run a Nautobot Job based on a single object as a source.
+    The button text field accepts Jinja2 template code to be rendered with an object as context.
+    """
+
+    content_types = models.ManyToManyField(
+        to=ContentType,
+        related_name="job_buttons",
+        verbose_name="Object types",
+        help_text="The object type(s) to which this job button applies.",
+    )
+    name = models.CharField(max_length=100, unique=True)
+    text = models.CharField(
+        max_length=500,
+        help_text="Jinja2 template code for button text. Reference the object as <code>{{ obj }}</code> such as <code>{{ obj.platform.slug }}</code>. Buttons which render as empty text will not be displayed.",
+    )
+    job = models.ForeignKey(
+        to="extras.Job",
+        on_delete=models.CASCADE,
+        help_text="Job this button will run",
+        limit_choices_to={"is_job_button_receiver": True},
+    )
+    weight = models.PositiveSmallIntegerField(default=100)
+    group_name = models.CharField(
+        max_length=50,
+        blank=True,
+        help_text="Buttons with the same group will appear as a dropdown menu. Group dropdown buttons will inherit the button class from the button with the lowest weight in the group.",
+    )
+    button_class = models.CharField(
+        max_length=30,
+        choices=ButtonClassChoices,
+        default=ButtonClassChoices.CLASS_DEFAULT,
+    )
+    confirmation = models.BooleanField(
+        help_text="Enable confirmation pop-up box. <span class='text-danger'>WARNING: unselecting this option will allow the Job to run (and commit changes) with a single click!</span>",
+        default=True,
```

Comment:
Can the default for this field be set to `False` since setting it to `True` is probably not the desired default setting considering how dangerous it can be?

Reply:
I believe you may be confusing `confirmation` with `commit`. The default here says all new Job Buttons will default to having a confirmation dialog pop-up modal rather than run via a single click.

There is no setting to enable or disable `commit` for the Job - it is always enabled.